src/resources/basis-worker.js

```diff
@@ -151,6 +155,35 @@ function BasisWorker() {
         return testInOrder(hasAlpha ? rgbaPriority : rgbPriority);
     };
 
+    // return true if the texture dimensions are valid for the target format
+    const dimensionsValid = (width, height, format, webgl2) => {
+        switch (format) {
+            // etc1, 2
+            case BASIS_FORMAT.cTFETC1:
+            case BASIS_FORMAT.cTFETC2:
+                // no size restrictions
+                return true;
+            // dxt1, 5
+            case BASIS_FORMAT.cTFBC1:
+            case BASIS_FORMAT.cTFBC3:
+                // width and height must be multiple of 4
+                return ((width & 0x3) === 0) && ((height & 0x3) === 0);
+            // pvrtc
+            case BASIS_FORMAT.cTFPVRTC1_4_RGB:
+            case BASIS_FORMAT.cTFPVRTC1_4_RGBA:
+                return isPOT(width, height) && ((width === height) || webgl2);
+            // astc
+            case BASIS_FORMAT.cTFASTC_4x4:
+                return true;
+            // atc
+            case BASIS_FORMAT.cTFATC_RGB:
+            case BASIS_FORMAT.cTFATC_RGBA_INTERPOLATED_ALPHA:
+                // TODO: remove atc support? looks like it's been removed from the webgl spec, see
+                // https://www.khronos.org/registry/webgl/extensions/rejected/WEBGL_compressed_texture_atc/
```

Comment:
This is really interesting - I had no idea that the ATC extension had been rejected. I wouldn't have a problem with removing it from the engine.

Reply:
It means we can remove ATC support from the basis wasm module. Not sure how much size that'll save us, but apparently 'every little helps'.